### ISSUE TITLE
No colored console-highlighter output when running as composer command

### ISSUE THREAD

**Issue by aleksblendwerk** [php-parallel-lint/PHP-Parallel-Lint#40]
When I run PHP-Parallel-Lint as a `composer` command, e.g. defined like this:

```
        "lint": "vendor/bin/parallel-lint --colors --exclude bin/.phpunit --exclude var --exclude vendor ."
```

there is no highlighting in the code snippets. There is a red "X" among the dots that indicate each checked file though, so it seems enforcing `--colors` does not apply to the highlighter output.

When I run the same command without `composer` the colors are present.

**Comment by esthezia** [comment]
I second this. Can this be resolved, please? It's been 5 years...

Maybe the mention here could help?: https://github.com/php-parallel-lint/PHP-Parallel-Lint/issues/42#issuecomment-783690946

Thank you!

**Comment by jrfnl** [comment]
As far as I can see, this is user-error. Please follow the [installation instructions](https://github.com/php-parallel-lint/PHP-Parallel-Lint#installation).

> For colored output, install the suggested package `php-parallel-lint/php-console-highlighter`:
>
>     composer require --dev php-parallel-lint/php-console-highlighter

If you still don't get coloured output, even when you have `php-console-highlighter` installed, please provide a detailed reproduction scenario.

**Comment by esthezia** [comment]
Yes, I have installed `php-console-highlighter`, and I can reproduce it.

```
codecheck/vendor/bin/parallel-lint --colors lbg
```
outputs:

<img width="884" height="251" alt="Image" src="https://github.com/user-attachments/assets/490754cd-f0eb-466e-a8ef-942bc01f2b2a" />

```
result="$(codecheck/vendor/bin/parallel-lint --colors lbg)"; echo "$result"
```
outputs:

<img width="884" height="252" alt="Image" src="https://github.com/user-attachments/assets/a5d2f946-ba21-47de-826b-583157bfc317" />

When you run it as a subcommand, even if it has the `--colors` flag, the code snippet is still not colorized.

**Comment by jrfnl** [comment]
Ah, so the problem is not necessarily PHP Parallel Lint, but the fact that you are running it via a subcommand ? In that case, I can't help you as I will not be able to reproduce this. Maybe someone else can.

**Comment by esthezia** [comment]
No, the problem is with PHP Parallel Lint. Whether ran as a subcommand or not, it should respect its flags, and colorize the code snippet if the `--colors` flag is passed.

As you can see, the "X" and the sentence "Syntax error found in 1 file" are red (red background), it's only the code snippet that is not coloured as it should be. This is a bug.

If you can run PHP Parallel Lint in bash, whether on Linux or in Windows via the Git Bash program, you can reproduce this.

**Comment by esthezia** [comment]
@jrfnl I think I have found a fix, and created a pull request: https://github.com/php-parallel-lint/PHP-Parallel-Lint/pull/190
Can you have a look and merge if it's ok?

Thank you!

**Comment by uuf6429** [comment]
I second this, I was incredibly confused after reading the README, it seemed to indirectly imply that colours will magically work after installing the 2nd package. A quick run showed that it only partially worked, exactly like @esthezia screenshot. So then I went to the other repo and I got even more confused when the usage section is just a snippet of code.

Usually one expects some form of configuration after installing a package, to say "hey, use this adapter/class/whatever to do the highlighting". I was about to report a ticket on improving the documentation, but luckily I thought about searching for issues, where I found this bug report..
